test/suite/test_checkpoint33.py

```diff
@@ -0,0 +1,108 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from test_cc01 import test_cc_base
+from suite_subprocess import suite_subprocess
+import time
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+
+# test_checkpoint33.py
+#
+class test_checkpoint33(test_cc_base, suite_subprocess):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,'
+    conn_config = 'cache_size=100MB,statistics=(all),statistics_log=(wait=1,json=true,on_close=true)'
+    uri = 'table:test_checkpoint33'
+
+    table_numkv = 10000
+    value_size = 1024
+    value = 'a' * value_size
+
+    def delete(self):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            c.set_key(k)
+            self.session.begin_transaction()
+            c.remove()
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(4))
+        c.close()
+
+    def populate(self):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            self.session.begin_transaction()
+            c[k] = self.value
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(2))
+        c.close()
+
+    def get_size(self):
+        c = self.session.open_cursor('statistics:' + self.uri, None, None)
+        file_size = c[stat.dsrc.block_size][2]
+        c.close()
+        return(file_size)
+
+    def test_checkpoint33(self):
+        # Pin oldest timestamp 1.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(1)}')
+
+        # Create and populate a table at timestamp 2.
+        self.session.create(self.uri, self.create_params)
+        self.session.checkpoint()
+        self.populate()
+
+        # Make everything stable at timestamp 3.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(3)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete everything at timestamp 4.
+        self.delete()
+
+        # Make the deletions stable at timestamp 5.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(5)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Make everything globally visible.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(5)}')
+        self.prout(f'File size: {self.get_size()}')
+
+        # Give checkpoint cleanup enough opportunity to clean up all the deleted pages.
+        cc_success = 0
+        while (cc_success < 10):
```

Comment:
Shouldn't it work the first time? If yes, we can use `wait_for_cc_to_run` from `text_cc01.py`

Reply:
I'm not sure why, but it doesn't work the first time, it takes a few iterations for checkpoint cleanup to cleanup everything. I'm not certain exactly how many passes are required or if that's deterministic but I found 10 to be more tha sufficient.

Reply:
@kommiharibabu, is it due to the limitation of pages that can be cleaned up per checkpoint?

Reply:
No. The new time window logic shouldn't affect the existing cleanup logic. The limitation is that it can only work for the Tw cleanup.

Reply:
I suggest to add eviction to evict all the pages to disk before they become obsolete can avoid the need for multiple checkpoints.

Reply:
Thanks @kommiharibabu, I've added that here: https://github.com/wiredtiger/wiredtiger/pull/11176/commits/3a8807caa7bd592ab577479cf49fee76f67eda09